### PR TITLE
i18n(es): add typescript page

### PR DESCRIPTION
**Pull request by kevinzunigacuellar**
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [ ] New or updated content
- [x] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

- adds `guides/typescript.md` page in Spanish 
- page is updated with changes in #761

### CODE REVIEW

File: src/pages/es/guides/typescript.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: TypeScript
+description: Aprenda a usar TypeScript incorporado en Astro.
+i18nReady: true
+---
+
+Astro tiene compatibilidad integrada con [TypeScript](https://www.typescriptlang.org/). Puedes importar archivos `.ts` y `.tsx` en su proyecto de Astro, e incluso escribir código TypeScript directamente dentro de su [componente de Astro](/es/core-concepts/astro-components/#script-del-componente).
```
Comment (agustinmulet) [inline src/pages/es/guides/typescript.md:8]:
```suggestion
Astro tiene compatibilidad integrada con [TypeScript](https://www.typescriptlang.org/). Puede importar archivos `.ts` y `.tsx` en su proyecto de Astro, e incluso escribir código TypeScript directamente dentro de su [componente de Astro](/es/core-concepts/astro-components/#script-del-componente).
```

File: src/pages/es/guides/typescript.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+Astro no realiza ninguna verificación de tipos por sí mismo. La verificación de tipos debe realizarse fuera de Astro, ya sea por su IDE o mediante un script separado. La [extensión de Astro VSCode](/es/editor-setup/) proporciona automáticamente sugerencias y errores de TypeScript en sus archivos abiertos.
```
Comment (agustinmulet) [inline src/pages/es/guides/typescript.md:10]:
```suggestion
Astro no realiza ninguna verificación de tipos por sí mismo. La verificación de tipos debe realizarse fuera de Astro, ya sea por el IDE o mediante un script separado. La [extensión de Astro VSCode](/es/editor-setup/) proporciona automáticamente sugerencias y errores de TypeScript en los archivos que posea abiertos.
```

File: src/pages/es/guides/typescript.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+## Configuración
+
+Se **recomienda encarecidamente** que cree un archivo `tsconfig.json` en su proyecto, para que las herramientas como Astro y VSCode sepan cómo entender su proyecto. Algunas funciones (como las importaciones de paquetes npm) no son totalmente compatibles con TypeScript sin un archivo `tsconfig.json`.
```
Comment (agustinmulet) [inline src/pages/es/guides/typescript.md:14]:
```suggestion
Se **recomienda encarecidamente** que cree un archivo `tsconfig.json` en su proyecto, para que las herramientas como Astro y VSCode sepan interpretarlo. Algunas funciones (como las importaciones de paquetes npm) no son totalmente compatibles con TypeScript a menos que crees un archivo `tsconfig.json`.
```

**Review by agustinmulet — COMMENTED** [review]
Some suggestions, let me know what you think!

File: src/pages/es/guides/typescript.md (hunk 1 above)
Comment (kevinzunigacuellar) [reply src/pages/es/guides/typescript.md:8]: he decidido cambiar el usted por tu para que suene un poco más familiar y amigable

**Comment by kevinzunigacuellar** [comment]
LGTM! 🚀

**Comment by Yan-Thomas** [comment]
Thanks @kevinzunigacuellar, @agustinmulet, great work!!

**Pull request merged**